src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -45,11 +50,18 @@ class ResolvedFormType implements ResolvedFormTypeInterface
 
     public function __construct(FormTypeInterface $innerType, array $typeExtensions = array(), ResolvedFormTypeInterface $parent = null)
     {
-        if (!preg_match('/^[a-z0-9_]*$/i', $innerType->getName())) {
+        $name = $innerType->getName();
+
+        // As of Symfony 2.8, getName() returns null by default. In this case
+        // (and as of Symfony 3.0), the FQCN of the type is used as name
+        // The deprecation error is triggered in FormRegistry
+        if (null === $name) {
+            $name = get_class($innerType);
+        } elseif (!preg_match('/^[a-z0-9_]*$/i', $name)) {
```

Comment:
don't think so. since getName is deprecated, we do not need to add new features for it.